src/p11_child/p11_child_openssl.c

```diff
@@ -582,6 +595,7 @@ errno_t init_p11_ctx(TALLOC_CTX *mem_ctx, const char *ca_db,
         return ENOMEM;
     }
 
+    ctx->ocsp_deadline = time(NULL) + timeout - 1;
```

Comment:
Hi,

you expect later that `ctx->ocsp_dealine` is `-1` if `timeout` is not given, so I think you have to check here as well if `timeout` is `-1`, i.e. not set.

Additionally and with respect to Alexey's comment, with `time(NULL) + timeout - 1` I think even a timeout of `0` make no sense, so it can be checked if `timeout > 0`.

bye,
Sumit

Reply:
Good catch. But even timeout 1s is too short. Deadline will be just now.

I think that we should increase the timeout before calling p11_child (i. e.  something like `if (timeout == 1) { DEBUG ("timeout 1s is too short"); timeout++ })`.  

What do you think?

Reply:
Hi,

I would like to suggest to do this in `p11_child` so that it can protect itself from a timeout too short. But otherwise I'm fine.

bye,
Sumit

Reply:
Hi,

thank you, the change looks good. Maybe it would be good to check the maximal value of `timeout` in `main()` of p11_child as well to make the implicit case from `long` to `time_t` a bit more safe. E.g. check if `timeout` is larger than `INT32_MAX` and if yes set it to `INT32_MAX`. What do you think?

bye,
Sumit

Reply:
done.

Reply:
Thanks, but I forgot that `timeout` is signed, so maybe limit negative values to `-1` as well?

Reply:
Thanks